src/lava/magma/core/model/py/model.py

```diff
@@ -148,37 +147,44 @@ def run(self):
                     if self.post_guard():
                         self.run_post_mgmt()
                     self.process_to_service_ack.send(MGMT_RESPONSE.DONE)
-                    # Handle VarPort requests from RefPorts
-                    if len(self.var_ports) > 0:
-                        self._handle_var_ports()
                 # Host phase - called at the last time step before STOP
                 elif enum_equal(phase, PyLoihiProcessModel.Phase.HOST):
-                    # Handle get/set Var requests from runtime service
-                    self._handle_get_set_var()
+                    pass
                 else:
                     raise ValueError(f"Wrong Phase Info Received : {phase}")
+            elif action == 'req':
+                # Handle get/set Var requests from runtime service
+                self._handle_get_set_var()
+            else:
+                # Handle VarPort requests from RefPorts
+                self._handle_var_port(action)
+
+            channel_actions = [(self.service_to_process_cmd, lambda: 'cmd')]
+            if enum_equal(phase, PyLoihiProcessModel.Phase.PRE_MGMT) or \
+                    enum_equal(phase, PyLoihiProcessModel.Phase.POST_MGMT):
+                for var_port in self.var_ports:
+                    for csp_port in var_port.csp_ports:
+                        if isinstance(csp_port, CspRecvPort):
+                            channel_actions.append((csp_port, lambda: var_port))
+            elif enum_equal(phase, PyLoihiProcessModel.Phase.HOST):
+                channel_actions.append((self.service_to_process_req,
```

Comment:
From a type perspective it is a bit inconsistent to give either a string or a port back as an action.